plugins/out_cloudwatch_logs/cloudwatch_api.c

```diff
@@ -520,12 +522,160 @@ int add_event(struct flb_cloudwatch *ctx, struct cw_flush *buf,
+    msgpack_sbuffer sbuf_emf;
+    msgpack_sbuffer_init(&sbuf_emf);
+
+    /* Serialize values into the buffer using msgpack_sbuffer_write */
+    msgpack_packer packer_emf;
+    msgpack_packer_init(&packer_emf, &sbuf_emf, msgpack_sbuffer_write);
+    int total_items = mk_list_size(flb_intermediate_metrics) + 1;
+    msgpack_pack_map(&packer_emf, total_items);
+
+    /* Pack the _aws map */
+    msgpack_pack_str(&packer_emf, 4);
+    msgpack_pack_str_body(&packer_emf, "_aws", 4);
+
+    msgpack_pack_map(&packer_emf, 2);
+
+    msgpack_pack_str(&packer_emf, 9);
+    msgpack_pack_str_body(&packer_emf, "Timestamp", 9);
+    msgpack_pack_long_long(&packer_emf, tms.tm.tv_sec * 1000L);
+
+    msgpack_pack_str(&packer_emf, 17);
+    msgpack_pack_str_body(&packer_emf, "CloudWatchMetrics", 17);
+    msgpack_pack_array(&packer_emf, 1);
+
+    msgpack_pack_map(&packer_emf, 3);
+
+    msgpack_pack_str(&packer_emf, 9);
+    msgpack_pack_str_body(&packer_emf, "Namespace", 9);
+
+    if (ctx->metric_namespace) {
+        msgpack_pack_str(&packer_emf, flb_sds_len(ctx->metric_namespace));
+        msgpack_pack_str_body(&packer_emf, ctx->metric_namespace, 
+                              flb_sds_len(ctx->metric_namespace));
+    }
+    else {
+        msgpack_pack_str(&packer_emf, 18);
+        msgpack_pack_str_body(&packer_emf, "fluent-bit-metrics", 18);
+    }
+
+    msgpack_pack_str(&packer_emf, 10);
+    msgpack_pack_str_body(&packer_emf, "Dimensions", 10);
+
+    struct mk_list *head, *inner_head;
+    struct flb_split_entry *dimension_list, *entry;
+    struct mk_list *csv_values;
+    if (ctx->metric_dimensions) {
+        msgpack_pack_array(&packer_emf, mk_list_size(ctx->metric_dimensions));
+
+        mk_list_foreach(head, ctx->metric_dimensions) {
+            dimension_list = mk_list_entry(head, struct flb_split_entry, _head);
+            csv_values = flb_utils_split(dimension_list->value, ',', 256);
+            msgpack_pack_array(&packer_emf, mk_list_size(csv_values));
+
+            mk_list_foreach(inner_head, csv_values) {
+                entry = mk_list_entry(inner_head, struct flb_split_entry, _head);
+                msgpack_pack_str(&packer_emf, entry->len);
+                msgpack_pack_str_body(&packer_emf, entry->value, entry->len);
+            }
+            flb_utils_split_free(csv_values);
+        }
+    }
+    else {
+        msgpack_pack_array(&packer_emf, 0);
+    }
+
+    msgpack_pack_str(&packer_emf, 7);
+    msgpack_pack_str_body(&packer_emf, "Metrics", 7);
+
+    if (strcmp(input_plugin, "cpu") == 0) {
+        msgpack_pack_array(&packer_emf, 3);
+    }
+    else if (strcmp(input_plugin, "mem") == 0) {
+        msgpack_pack_array(&packer_emf, 6);
+    }
+    else {
+        msgpack_pack_array(&packer_emf, 0);
+    }
+
+    mk_list_foreach_safe(metric_head, metric_temp, flb_intermediate_metrics) {
+        struct flb_intermediate_metric *an_item = mk_list_entry(metric_head, 
```

Comment:
declare variables on top of the function

Reply:
Updated.